eyre/src/error.rs

```diff
@@ -116,6 +116,7 @@ impl Report {
         unsafe { Report::construct(error, vtable, handler) }
     }
 
+    #[cfg(feature = "anyhow")]
```

Comment:
Is the removal of `from_display` intended, as it is useful of conversion of non-error errors such as string or other display structs.

I've run into this for `no-std` crates, which do not implement `std::error::Error` for their exported error type, but rather just `Display + Debug`

Reply:
`from_display` is `pub(crate)` and only used in the `anyhow` compatibility layer.

I would think changing it to `pub` is outside the scope of this PR.

Reply:
Ok got it, thought it was also used by the macro, but it used the `from_adhoc` instead